### PR TITLE
add apm-server helm chart

### PR DESCRIPTION
**Pull request by pbecotte**
- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [x] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

Take a stab at adding the apm-server helm chart. Not sure about an open source image, or if there's a good way to flesh out the `goss` tests. I am NOT an apm-server expert, any suggestions are welcome. I have this running on my cluster currently.

### CODE REVIEW

**Comment by jmlrt** [comment]
related to https://github.com/elastic/helm-charts/issues/68

**Comment by jmlrt** [comment]
Hi @pbecotte,
Thank you for this great PR!
Adding yourself Goss tests and Python tests is really nice 👍.
The first think I see is that `httpPort` is missing in your `values.yml`.
We'll do a full review soon.

File: apm-server/values.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
+---
+# Comma separated list of hosts
```
Comment (jmlrt) [inline apm-server/values.yaml]: `httpPort` is missing

File: apm-server/examples/default/test/goss.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+port:
+  tcp:8200:
```
[the remark this answers is not in the document]
Comment (pbecotte) [reply apm-server/examples/default/test/goss.yaml]: Seems likely- but I don't really know that much about apm-server to write more useful tests... any ideas?

**Comment by pbecotte** [comment]
Addressed your comments- not sure how I had so many bugs in there- the tests passed at some point, but apparently I didn't run them before pushing? Not sure what would be a good test for apm server- maybe there is a health check url we should send a request to?

File: apm-server/examples/default/test/goss.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmlrt) [reply apm-server/examples/default/test/goss.yaml]: For Logstash Helm Chart in #347 we decided to drop the `port` test and use a [http test](https://github.com/jmlrt/helm-charts/blob/master/logstash/examples/default/test/goss.yaml#L7-L19).

You could use this `http` test for APM (I don't know if there is a more relevant endpoint)

```
http:
  http://localhost:8200?pretty:
    status: 200
    timeout: 2000
    body:
      - '"version" : "7.4.0"'
```

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
Hi @pbecotte,

Better late than never, we are thinking about merging this PR before the end of the month and release APM server chart in alpha for 7.6.0.

I think we still had a few comments in https://github.com/elastic/helm-charts/pull/324#pullrequestreview-315903104 and certainly a few others to come in the next days.

Are you still interested to work on it?

**Comment by pbecotte** [comment]
Yeah, didn't want to get too far out in front of it, but would be happy to go through and update it.

File: apm-server/README.md [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+## Requirements
+
+* Kubernetes >= 1.9
+* [Helm](https://helm.sh/) >= 2.8.0
+
+## Usage notes and getting started
+* The default APM Server configuration file for this chart is configured to use an 
+Elasticsearch endpoint as configured by the rest of these helm charts. This can
+easily be overridden in the config value apmConfig.apm-server.yml
+
```
[the remark this answers is not in the document]
Comment (jmlrt) [reply apm-server/README.md]: When I mentionned a note, it was a bullet point like in other [README](https://github.com/elastic/helm-charts/blame/master/logstash/README.md#L15) (sorry for not being enough explicit).

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Pull request merged**